src/plugins/node-workspace.ts

```diff
@@ -14,33 +14,48 @@
 
 import * as semver from 'semver';
 import cu = require('@lerna/collect-updates');
-import {Package, PackageJson} from '@lerna/package';
+import {Package as LernaPackage, PackageJson} from '@lerna/package';
 import {PackageGraph, PackageGraphNode} from '@lerna/package-graph';
 import {runTopologically} from '@lerna/run-topologically';
 import {ManifestPlugin} from './plugin';
 import {ManifestPackageWithPRData, ManifestPackage} from '..';
 import {VersionsMap} from '../updaters/update';
-import {packageJsonStringify} from '../util/package-json-stringify';
+import {jsonStringify} from '../util/json-stringify';
 import {CheckpointType} from '../util/checkpoint';
 import {RELEASE_PLEASE} from '../constants';
 import {Changes} from 'code-suggester';
 import {ConventionalCommits} from '../conventional-commits';
 import {Changelog} from '../updaters/changelog';
 
-type PathPkgJson = Map<string, PackageJson>;
+class Package extends LernaPackage {
```

Comment:
What do you think about something like this?

Reply:
sorry for the lag - I think I get this approach and it seems good to me. but I need to spend another minute reading it all the way through